Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractRandomCutTree.java

```diff
@@ -376,7 +424,8 @@ void resolve(Point point, NodeReference parentIndex,
                 Cut cut = randomCut(random, mergedBox);
                 // the cut is between a leaf node and the new point; it must exist
                 AddPointState<Point, NodeReference, PointReference> newState = new AddPointState<>(pointIndex);
-                newState.initialize(nodeReference, cut.getDimension(), cut.getValue(), sequenceNumber, mergedBox);
+                newState.initialize(nodeReference, cut.getDimension(), cut.getValue(), sequenceNumber, mergedBox,
```

Comment:
I would consider dropping the initialize method, since it's just setting fields, and setting the fields directly in the method instead.

```
AddPointState<Point, NodeReference, PointReference> state = new AddPointState<>(pointIndex);
state.setNodeReference(nodeReference);
state.setCutDimension(cut.getDimensions());
// etc.
```

I guess what `initialize` gets us is that it won't let you forget to set one of the arguments, and it turns 6 or 7 lines of repeated code into a single long method invocation. I don't feel too strongly about this one.

Although, I do wonder: why is `pointIndex` passed to the constructor, but everything else is set by `initialize`?

Reply:
The case of pointIndex being passed to constructor is for an exact duplicate. We need that state to be returned. The other stuff is basically context. I preferred setting the 4 in a single step because they are interdependent.